src/AgentImplementation.sol

```diff
@@ -175,8 +195,16 @@ contract AgentImplementation is IAgent, ERC721Holder, ERC1155Holder {
                 wrappedAmount = _getBalance(wrappedNative);
             }
 
-            // Set _callback who should enter one-time execute
-            if (logics[i].callback != address(0)) _caller = logics[i].callback;
+            // Set callback who should enter one-time `executeByCallback`
+            if (logics[i].callback != address(0)) {
+                bytes32 callback = bytes32(bytes20(logics[i].callback));
+                if (shouldChargeFeeByLogic) {
+                    // Set the least significant bit
+                    _callbackWithCharge = callback |= _CHARGE_MASK;
+                } else {
```

Comment:
Do we only need to update the `_callbackWithCharge` function and not the `callback` variable?
```suggestion
              _callbackWithCharge = callback | _CHARGE_MASK;
```